source/docs/advanced/theme/wysiwyg.md

```diff
@@ -4,116 +4,385 @@ title: Display WYSIWYG content
++  CustomWysiwyg: loadable(() => import("./CustomWysiwyg"))
+};
+```
+
+This is it, you have a new `CustomWysiwyg` type. However, the components rendered are still simple HTML tags. If you want to add interactivity, you will need to refer to the `theme/modules/WysiwygV2/CustomWysiwyg` you have created.
+
+In this file you will notice that there is a `defaultComponentsMap` constant. This object is used to reference how to render each html tag in your Wysiwyg component. We call these **transforms**: they transform an html tag into a React component. This is what will allow you to finely tune how your Wysiwyg is displayed and add interactivity to your content.
+
+For instance, if you want your `h2` tags to have some additional classes, you could use the following `defaultComponentsMap` instead:
 
 ```js
-{
-  // regex to identify the shortcode within your HTML code
-  regex: /\{\{media url="(.*?)"\}\}/gi,
-  // How to replace the given shortcode
-  replacement: (match, url) => `/media/${url}`
+const defaultComponentsMap = {
+  h2: (props) => <h2 {...props} className="wysiwyg-h2" />
 }
 ```
 
-See [EnhanceWysiwyg's reference](/docs/reference/wysiwyg.html) for more details.
+The props available represent the HTML attributes coming from your Wysiwyg content. The only difference compared to traditional HTML is that `class` is renamed to `className` to better support React patterns and the `props.children` is an already constructed React element.
+
+<blockquote class="important">
+<b>Note:</b> If the component is complex, please think about code splitting them to avoid too large bundle on pages that don't need them.
+</blockquote>
 
-### Transforms
+#### Fetching data to render Wysiwyg components
 
-Transforms are functions that take an HTML node and transform it in a React Element.
+You can now transform HTML tags into React components. However, in some cases the data provided in the raw HTML you've received from your remote service won't be enough to display your component. For instance, you may have a `<product-name sku="VSK12" />` that is supposed to display the product's name. But all you have in the attributes is the SKU. This can be done by associating a `WysiwygNodeData` with your HTML node.
 
-For instance, we want to make sure that it transforms `<a>` tags to `Link` elements to preserve React's navigation. For instance, `<a href="/about-us">About us</a>` will be transformed to `<Link to="/about-us">About us</Link>`. It looks like this:
+Indeed, when you've created your `CustomWysiwyg` type, you have declared: `childNodes` (`JSON`) and `data` (`[WysiwygNodeData]`). By default, `data` will always be empty. But you can tell the Wysiwyg loader to fetch data for some specific `childNodes` (that each represent an HTML tag). This means that for `product-name`, you will associate a new `data` element that will contain the whole product data needed to display the component. To do so, you need to follow these steps:
```

Comment:
The transition is misleading as readers may think that these types are about client-side code.
```suggestion
Indeed, when you've created your GraphQL `CustomWysiwyg` type, you have declared: `childNodes` (`JSON`) and `data` (`[WysiwygNodeData]`). By default, `data` will always be empty. But you can tell the Wysiwyg loader to fetch data for some specific `childNodes` (that each represent an HTML tag). This means that for `product-name`, you will associate a new `data` element that will contain the whole product data needed to display the component. To do so, you need to follow these steps:
```

Ideally, we could also transform "you've created your GraphQL `CustomWysiwyg` type" into a link to the correct anchor.